documentation/API_documentation/QoD_Stable_Latency_API.md

```diff
@@ -0,0 +1,181 @@
+Following table provides an overview of common error names, codes and messages applicable to QoD API.
+
+| No | Error Name | Error Code | Error Message |
+| --- | ---------- | ---------- | ------------- |
+| 1 | Invalid port(s) | 400 | "Ports specification not valid |
+| 2 | Invalid protocol | 400 | "Validation failed for parameter: protocol" |
+| 3 | Invalid QoS profile | 400 | "Validation failed for parameter: QoS-profile" |
+| 4 | Invalid IP address (format) | 400 | "Validation failed for parameter: IP-addr" |
+| 5 | Invalid duration | 400 | "Validation failed for parameter: Session duration" |
+| 6 | Unauthorized | 401 | "Un-authorized to invoke operation" |
+| 7 | Forbidden | 403 | "Forbidden to invoke operation" |
+| 8 | Session with same parameters already exists | 409 | "Found session \<session> already active until \<expirationTime>" |
+| 9 | Service unavailable | 503 | “Internal error due to requrired telco service unvailability" |
+
+### 4.4 Policies
+
+N/A
+
+### 4.5 Code Snippets
+<br>
+<span class="colour" style="color:rgb(36, 41, 47)">Snippet 1, elaborates REST based API call with "*curl"* to create a QoS session for sample streaming service with following parameters: </span>
+
+* Latency QoS session with 1H duration and QoS-profile "LOW\_Latency" mapping,
+* App-Flow is specified for UDP protocol with UE-Terminal IP address (ueAddr=10.0.0.1), Application server network (asAddr=54.204.25.0/28) and Port number (asPorts=33001).
+
+Please note, the credentials for API authentication purposes need to be adjusted based on target security system configuration.
+
+| Snippet 1. Create QoS session to manage latency |
+| ----------------------------------------------- |
+| curl -X 'POST' '[https://\[replace](https://%5Breplace) with your base URL]/qos-senf/v1/sessions'   <br>    -H 'accept: application/json' <br>    -H 'Content-Type: application/json'<br>    -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbG...."<br>    -d '{<br>     "duration": 3600,<br>     "ueAddr": "10.0.0.1",<br>     "asAddr": "54.204.25.0/28",<br>     "asPorts": "33001",<br>     "protocolOut": "UDP",<br>     "qos": "LOW\_Latency",<br>     "notificationUri": "[https://your-callback-server.com/notifications](https://your-callback-server.com/notifications)",<br>     "notificationAuthToken": "c8974e592c2fa383d4a3960714"<br>   }' |
+<br>
+Snippet 2, elaborates sample QoS notification "SESSION\_TERMINATION" message distributed from QoD backend to client callback function.
+
+| Snippet 2. Sample QoS session notification |
+| ------------------------------------------ |
+| {<br>   "sessionId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",   "event": "SESSION\_TERMINATED"<br>} |
+
+### 4.6 FAQ's
+
+* Q: *My application server components are distributed between multiple public IP addresses. In this case how the asAddr parameter should be defined?*
+A: You can use the network segment for asAddr. Currently there are no restrictions to the size of the network segment.
```

Comment:
Instead of 'network segment' we should be specific, e.g. CIDR notation with network mask

Reply:
Taken FAQs out of current PR, rework needed.